Material

osunac opened the pull request:
Here are some commits to give PReVo a more modern look. I would like to add a few more (e.g. white background) but I need to know if current patches might be merged before going too far.

Changes include:
- Gradle support (required for Android Studio)
- Increase minimum SDK to version 8 (Android 2.2)
- Use AppCompat library. It is possible that this change breaks build with Ant
- Removal of _About_ screen
- Replace zoom buttons by gesture
- Replace search button with Floating Actiion Button. Some people discourage the use of that kind of button, but it makes searching easier.
- More space
- Hide toolbar when scrolling
- New icon - the idea was to have something closer to material design guidelines but it's not very beautiful.

osunac committed:
Add Gradle support

This makes it easier to work with Android Studio.

osunac committed:
Remove About diaglog

About dialog is not standard in Android applications.

osunac committed:
Migrate to AppCompat library

Required for Material Design.

Increase min SDK to 8
 - required for Log.wtf()
 - required for gestures (later commit)
 - devices with Android 2.1 will not be able to migrate
Set target SDK to 23
Replace Activity with AppCompatActivity
Theme inherits from Theme.AppCompat
XML uses 'app' namespace where required

Reference:
https://guides.codepath.com/android/Migrating-to-the-AppCompat-Library

osunac committed:
Replace search menu entry by floating action button

Reference: https://guides.codepath.com/android/Floating-Action-Buttons

osunac committed:
Increase readability

Article view margins
Add space before section headers.
Remove search language in article activity (already set in search activity).
Label update.

osunac committed:
Replace zoom buttons by scale gesture

osunac committed:
Replace ListActivity with AppCompatActivity

osunac committed:
Replace action bar by Toolbar

This is required since the selected theme has no action bar.

In the article view, the toolbar will disappear when scrolling down.

osunac committed:
New launcher icon

spagy commented:
@bpeel, if the merge conflicts were to be resolved, is there anything else stopping this pull request from being merged?

bpeel commented:
I’m sorry for not replying to this pull request sooner. I think it is not ready to land. Some things that put me off the request are:

- The branch is a collection of various unrelated changes. It would probably make sense to have separate requests for each of them.
- I don’t want to increase the minimum Android version unless there is a really good reason for it.
- The patches make a lot of unrelated coding style changes which make them difficult to review.
- I don’t want to remove the about dialog without having another way to check the version number and display the mandatory GPL copyright message.
- In my opinion the proposed new icon looks worse.

osunac commented:
Thanks for the feedback. I will create various pull requests addressing the issues when I can spare some time. At least you will know where the requests are headed to.
